src/components/modals/load-storage-map.js

```diff
@@ -262,21 +277,38 @@ function LoadStorageMapFactory() {
         provider
       });
     }
+
+    _clickBack = () => {
+      this.setState({showProviderSelect: true});
+    };
+
+    _selectProvider = providerName => {
+      this.props.onSetCloudProvider(providerName);
+      const provider = (this.props.cloudProviders || []).find(p => p.name === providerName);
```

Comment:
what if provider is null?

I know we have a check in getSavedMaps but should we even dispatch the action if provider is undefined

Reply:
`_selectProvider` is callbacks from `ProviderSelect` which is rendered by cloudProviders. I don't  see in which case provider will be underfined